# Advent of Code Day 1!

### Part 1: Distance between two lists

In [1]:
import numpy as np
import pandas as pd

In [ ]:
# read in data
input = pd.read_csv('../data/day01_input.txt', header = None, delimiter=' ')

In [ ]:
# subset the columns
first = input.iloc[:, 0].copy()
second = input.iloc[:, 3].copy()

In [ ]:
# sort in ascending order
first = first.sort_values()
second = second.sort_values()

In [ ]:
# make lists
first = list(first)
second = list(second)

In [ ]:
# calculate distance with lost comprehension
dist = [abs(f - s) for f, s in zip(first, second)]

In [ ]:
# add all terms for reduced metric
sum(dist)

1258579

**Final Answer:** 1258579

### Part 2: Similarity score between two lists

In [45]:

# map and multiply to create similarity scores
sum(map(lambda a: a * second.count(a), first))



23981443

**Final Answer:** 23981443